src/mui/input/CheckboxGroupInput.js

```diff
@@ -0,0 +1,123 @@
+ * @example
+ * const choices = [
+ *    { _id: 123, full_name: 'Leo Tolstoi', sex: 'M' },
+ *    { _id: 456, full_name: 'Jane Austen', sex: 'F' },
+ * ];
+ * <CheckboxGroupInput source="author_id" choices={choices} optionText="full_name" optionValue="_id" />
+ *
+ * `optionText` also accepts a function, so you can shape the option text at will:
+ * @example
+ * const choices = [
+ *    { id: 123, first_name: 'Leo', last_name: 'Tolstoi' },
+ *    { id: 456, first_name: 'Jane', last_name: 'Austen' },
+ * ];
+ * const optionRenderer = choice => `${choice.first_name} ${choice.last_name}`;
+ * <CheckboxGroupInput source="author_id" choices={choices} optionText={optionRenderer} />
+ *
+ * `optionText` also accepts a React Element, that will be cloned and receive
+ * the related choice as the `record` prop. You can use Field components there.
+ * @example
+ * const choices = [
+ *    { id: 123, first_name: 'Leo', last_name: 'Tolstoi' },
+ *    { id: 456, first_name: 'Jane', last_name: 'Austen' },
+ * ];
+ * const FullNameField = ({ record }) => <span>{record.first_name} {record.last_name}</span>;
+ * <CheckboxGroupInput source="gender" choices={choices} optionText={<FullNameField />}/>
+ *
+ * The object passed as `options` props is passed to the material-ui <RadioButtonGroup> component
+ */
+export class CheckboxGroupInput extends Component {
+    handleChange = (event, value) => {
+        console.log(this.props.input.value);
+        this.props.input.onChange(value);
+    }
+
+    render() {
+        const { input, label, source, choices, optionText, optionValue } = this.props;
+        const option = React.isValidElement(optionText) ? // eslint-disable-line no-nested-ternary
+            choice => React.cloneElement(optionText, { record: choice }) :
+            (typeof optionText === 'function' ?
+                optionText :
+                choice => choice[optionText]
+            );
+
+        return (
+            <Labeled label={label} onChange={this.handleChange} source={source}>
+                <div>
+                    {choices.map((choice, index) => (
+                        <input
+                            key={choice[optionValue]}
+                            type="checkbox"
+                            checked={input.value.indexOf(choice[optionValue]) >= 0}
+                            onChange={event => {
+                                const index = input.value.indexOf(choice[optionValue]);
+                                if (index < 0) { // wasn't selected
+                                    if(event.target.checked) { // was checked
+                                        input.onChange(input.value.concat(choice[optionValue]));
+                                    }
+                                } else {
+                                    if(event.target.checked) { // was unchecked
+                                        const copy = [...input.value]; // make copy to not mutate value
```

Comment:
So you're copying the entire array... to mutate it. I'd prefer a double slice, as in [this reducer](https://github.com/marmelab/admin-on-rest/blob/97bc173a3de3d0d3ddedc2fa8961c58e7510ebbe/src/reducer/resource/list/ids.js#L11-L15);